### PR TITLE
Migrate xslt native functions to java interoperability

### LINKED ISSUE #20119: [xslt] Migrate native functions to java interoperability
**Description:**
$title

### PR DESCRIPTION
**Pull request by aashikam**
**BLOCKED DUE TO:** https://github.com/ballerina-platform/ballerina-lang/issues/20219 

Fixes #20119

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/xslt/src/main/ballerina/src/xslt/natives.bal [hunk 1]
```diff
@@ -14,9 +14,14 @@
...
 
+import ballerinax/java;
+
 # Transforms the single-rooted XML content to another XML/HTML/plain text using XSL transformations.
 #
 # + input - An XML object, which needs to be transformed
 # + xsl - The XSL style sheet represented in an XML object
 # + return - The transformed result represented in an XML object or the `error` in case of an error
-public function transform(xml input, xml xsl) returns xml|error = external;
+public function transform(xml input, xml xsl) returns xml|error =
```
Comment (ThisaruGuruge) [inline stdlib/xslt/src/main/ballerina/src/xslt/natives.bal]: Shall we create a separate __*private*__ inter-op function, and call it inside the __*public*__ function ?

Comment (wggihan) [reply stdlib/xslt/src/main/ballerina/src/xslt/natives.bal]: Why we need to create intermediated  function?

Comment (ThisaruGuruge) [reply stdlib/xslt/src/main/ballerina/src/xslt/natives.bal]: It is the practice we had during inter-op migration. The intention is to keep the public function as a pure ballerina function.

Comment (wggihan) [reply stdlib/xslt/src/main/ballerina/src/xslt/natives.bal]: IINM,  we had to use intermediated function if this migration causing any API change due to that. Specially arguments with string types, we had to use it. But if exisiting API can support without breaking the backwards compatibility the we don't need another function to bridge IMO.

Comment (ThisaruGuruge) [reply stdlib/xslt/src/main/ballerina/src/xslt/natives.bal]: +1.

**Pull request merged**